pkg/service/v0/graph.go

```diff
@@ -38,14 +40,12 @@ func (g Graph) UserCtx(next http.Handler) http.Handler {
 		if userID := chi.URLParam(r, "userID"); userID != "" {
 			user, err = g.ldapGetUser(userID)
 		} else {
-			// TODO: we should not give this error out to users
-			// http.Error(w, err.Error(), http.StatusInternalServerError)
-			render.Status(r, http.StatusNotFound)
+			errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest)
 			return
 		}
 		if err != nil {
 			g.logger.Info().Msgf("error reading user: %s", err.Error())
```

Comment:
You got a structured logger, so please use this structure....

```
g.logger.Info().
	Err(err).
	Msg("Failed to read user")
```

Beside that, for logging errors like the way you did you should always write something like `Msgf("error reading user: %w", err)`.